src/common/utils/compare.ts

```diff
@@ -10,30 +17,201 @@ export interface RankedLocationSummary extends SummaryForCompare {
   rank: number;
 }
 
+export const orderedMetrics = [
+  Metric.CASE_DENSITY,
+  Metric.CASE_GROWTH_RATE,
+  Metric.POSITIVE_TESTS,
+  Metric.HOSPITAL_USAGE,
+  Metric.CONTACT_TRACING,
+];
+
 const locations: any = getLocationNames();
 
-export function getStatesArr(): SummaryForCompare[] {
+function getLocationObj(locationInfo: Location) {
+  if (locationInfo.full_fips_code) {
+    return {
+      locationInfo: locationInfo,
+      metricsInfo:
+        locationInfo.full_fips_code &&
+        countySummary(locationInfo.full_fips_code),
+    };
+  } else {
+    return {
+      locationInfo: locationInfo,
+      metricsInfo: stateSummary(locationInfo.state_code),
+    };
+  }
+}
+
+export function getAllStates(): SummaryForCompare[] {
   return locations
     .filter((location: Location) => !location.county)
-    .map((stateInfo: Location) => {
-      return {
-        locationInfo: stateInfo,
-        metricsInfo: stateSummary(stateInfo.state_code),
-      };
-    });
+    .map(getLocationObj);
 }
 
-export function getCountiesArr(stateId: string): SummaryForCompare[] {
+export function getAllCounties(): SummaryForCompare[] {
+  return locations
+    .filter((location: Location) => location.county)
+    .map(getLocationObj);
+}
+
+export function getAllCountiesOfState(stateId: string): SummaryForCompare[] {
   return locations
     .filter(
       (location: Location) =>
         location.county && location.state_code === stateId,
     )
-    .map((countyInfo: Location) => {
-      return {
-        locationInfo: countyInfo,
-        metricsInfo:
-          countyInfo.full_fips_code && countySummary(countyInfo.full_fips_code),
-      };
-    });
+    .map(getLocationObj);
+}
+
+// TODO (chelsi) filter-related WIP:
+// consolidate location and national MSA utils?
+export function getAllMetroCounties(): SummaryForCompare[] {
+  return locations
+    .filter(
+      (location: Location) =>
+        location.county &&
+        location.full_fips_code &&
+        getCountyMsaCode(location.full_fips_code),
+    )
+    .map(getLocationObj);
+}
+
+export function getAllNonMetroCounties(): SummaryForCompare[] {
+  return locations
+    .filter(
+      (location: Location) =>
+        location.county &&
+        location.full_fips_code &&
+        !getCountyMsaCode(location.full_fips_code),
+    )
+    .map(getLocationObj);
+}
+
+export function getNeighboringCounties(
+  countyFips: string,
+): SummaryForCompare[] {
+  const neighboringCountiesFips = getAdjacentCounties(countyFips);
+  return locations
+    .filter(
+      (location: Location) =>
+        location.full_fips_code &&
+        location.full_fips_code !== countyFips &&
+        neighboringCountiesFips.includes(location.full_fips_code),
+    )
+    .map(getLocationObj);
```

Comment:
Just a reminder to add the county to the list of neighboring counties! I'm not sure if that belongs here or in the table before passing the locations to the table though.